### ISSUE TITLE
Prerenderer silently fails on PhantomJS crashes.

### ISSUE THREAD

**Issue by MrOpposite** [prerender/prerender#378]
I have logger enabled and had to put in a lot of effort in finding why my SPA was rendering just fine in my browser, but Prerender didn't even seem to execute any javascript.

After reducing the code that crashes to three lines, I tried to run it trough Phantom and got this error:
```
phantomjs> (function (){'use strict';console.log('hi');var a = function b (b) {};})();
Cannot declare a parameter named 'b'  in strict mode

  phantomjs://repl-input:1 in global code
```

I've already submitted an issue about [that particular error](https://github.com/ariya/phantomjs/issues/14778) to the phantomjs repo. Feel free to poke the devs to fix this issue if you'd like!

Having that error printed into the console from the Prerenderer would greatly have helped with nailing this issue down.

When this error occured, Prerenderer did not execute any JS at all, so the only way to check whether it was executing or no was to place a `console.log('hi');` at the start of the code and start removing random bits until 'hi' appeared in the prerenderer output, and then start trying to figure out how to find out what's wrong. Not a great way of debugging I'm afraid.

So yeah, if you could fix so that I'd receive these kind of errors from Prerender stdout or stderr I'd be very happy!

**Comment by MrOpposite** [comment]
Actually, after some research, it seems like this is on PhantomJS, not Prerender. The onError-callback is not called due to JS-parsing errors, and, for some weird reason, this specific strict-mode error. Other strict mode errors work fine like for example this
```
"use strict";
a = 4;
```

Which correctly produces 
```
2017-01-25T10:33:55.107Z getting http://dobby/
ReferenceError: Strict mode forbids implicit creation of global property 'a'

  http://dobby/js/app.js:3 in global code
```

And from my own testing, I can't seem to be able to get either parsing errors nor the Cannot redeclare-errors from the phantomjs page-object.

You can probably safely close this issue. I'll be throwing this issue at PhantomJS instead.

**Comment by thoop** [comment]
Yeah we've seen this before with other strict mode errors. I'm hoping it's fixed in the latest PhantomJS 2.5 that is about to be released (currently in beta).

Sorry for the inconvenience there :(
